src/main/java/org/benetech/servicenet/repository/ProviderRecordsRepository.java

```diff
@@ -225,6 +312,14 @@ private Long getTotal(ProviderFilterDTO providerFilterDTO, UserProfile userProfi
         return em.createQuery(countCriteria).getSingleResult();
     }
 
+    private Long getTotal(ProviderFilterDTO providerFilterDTO, Silo silo, String search) {
+        CriteriaQuery<Long> countCriteria = cb.createQuery(Long.class);
+        Root<Organization> selectRoot = countCriteria.from(Organization.class);
+        this.addFilters(countCriteria, selectRoot, silo, providerFilterDTO, search);
+        countCriteria.select(cb.countDistinct(selectRoot));
+        return em.createQuery(countCriteria).getSingleResult();
```

Comment:
I think that when em.createQuery(countCriteria) will be empty, then .getSingleResult() will throw nullpointer. Same case as it happened here: https://github.com/benetech/ServiceNet/blob/dev/src/main/java/org/benetech/servicenet/repository/ShelterRepositoryImpl.java#L101